source/common/network/lc_trie.h

```diff
@@ -42,7 +42,7 @@ class LcTrie {
    *                              factor. It reduces the depth of the trie.
    */
   LcTrie(const std::vector<std::pair<std::string, std::vector<Address::CidrRange>>>& tag_data,
-         double fill_factor = 0.5, uint32_t root_branching_factor = 0);
+         double fill_factor = 0.5, uint32_t root_branching_factor = 16);
```

Comment:
I think the `root_branching_factor` should be 4 or 8, rather than 16. A value of 16 will result in the trie having at least 2^16 nodes, which will be orders of magnitude bigger than the number of prefixes in a typical proxy configuration. As far as I can tell from the original LC Trie paper, Nilsson and Karlsson seem to have been targeting a different use case: routers configured with hundreds of thousands of prefixes.

Reply:
Hmm I see your point. 

A simpler option is to let the trie calculate the value itself. All of these values(4,8,16) are a bit of a guess. Once we have perf numbers, then this can be tweaked.